.github/workflows/nuget-publish.yml

```diff
@@ -0,0 +1,70 @@
+name: NuGet Publish
+
+on:
+  push:
+    branches: [ master ]
+
+env:
+  NUGET_DIR: ${{ github.workspace }}/nuget
+  NUGET_ARTIFACT: nuget
+
+jobs:
+  testing:
+    if: contains(github.event.head_commit.message, 'chore(release)') 
+    uses: ./.github/workflows/test.yml
+
+  pack: 
+    if: contains(github.event.head_commit.message, 'chore(release)') 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+
+      - run: sed -E 's:RepositoryCommit>.+<\/:RepositoryCommit>${{ github.sha }}<\/:g' ButterCMS/ButterCMS.csproj > tmp && mv tmp ButterCMS/ButterCMS.csproj
+
+      - run: dotnet pack --configuration Release --output ${{ env.NUGET_DIR }}
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.NUGET_ARTIFACT }}
+          path: ${{ env.NUGET_DIR }}
+          if-no-files-found: error
+          retention-days: 1
+
+  validate-nuget:
+    runs-on: ubuntu-latest
+    needs: pack
+    steps:
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ env.NUGET_ARTIFACT }}
+          path: ${{ env.NUGET_DIR }}
+
+      - name: Install nuget validator
+        run: dotnet tool update Meziantou.Framework.NuGetPackageValidation.Tool --global
+
+      - name: Validate nuget package
+        run: meziantou.validate-nuget-package $(ls ${{ env.NUGET_DIR }}/*.nupkg | head -n 1) --excluded-rules Symbols
```

Comment:
1. What are possible files that `ls ${{ env.NUGET_DIR }}/*.nupkg` will find?
2. Are we sure that `ls ${{ env.NUGET_DIR }}/*.nupkg | head -n 1` (the first file in the list) will _always_ evaluate to the exact file we want? IIRC it will be the first file in the lexicographical order.
3. [shellcheck](https://www.shellcheck.net/) finds a few issues with the code:

```
$ shellcheck -s bash a.sh

In a.sh line 1:
meziantou.validate-nuget-package $(ls ${{ env.NUGET_DIR }}/*.nupkg | head -n 1) --excluded-rules Symbols
                                 ^-- SC2046 (warning): Quote this to prevent word splitting.
                                   ^-----------------------------^ SC2012 (info): Use find instead of ls to better handle non-alphanumeric filenames.
                                      ^-----------------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                                        ^--------------^ SC2296 (error): Parameter expansions can't start with {. Double check syntax.
                                                         ^-- SC1083 (warning): This } is literal. Check expression (missing ;/\n?) or quote it.

Did you mean:
meziantou.validate-nuget-package $(ls "${{ env.NUGET_DIR }"}/*.nupkg | head -n 1) --excluded-rules Symbols

For more information:
  https://www.shellcheck.net/wiki/SC2296 -- Parameter expansions can't start ...
  https://www.shellcheck.net/wiki/SC1083 -- This } is literal. Check expressi...
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...
```

You may consider fixing them (excluding the `SC2296` one, which I believe is caused by double `{` in `${{ env.NUGET_DIR }}`, which is a valid syntax for variable substitution in GH actions).

Reply:
@Toreno96 There should be only one file in `${{ env.NUGET_DIR }}/*.nupkg` unless someone change packing process.  However, I changed the line to 

```bash
for FILE in ${{ env.NUGET_DIR }}/*.nupkg; do meziantou.validate-nuget-package "$FILE" --excluded-rules Symbols; done
```

so no file is going to be ignored. The same for the push. 

Also checked commands with **shellcheck**.

Reply:
> There should be only one file

OK, but what's the example name of this file?

Reply:
Cannot we simplify to:
```
$ meziantou.validate-nuget-package ${{ env.NUGET_DIR }}/*.nupkg --excluded-rules Symbols
```
??

Does `meziantou.validate-nuget-package` not accept multiple files at once?

Reply:
> > There should be only one file
> 
> OK, but what's the example name of this file?

`ButterCMS.2.0.0.nupkg`

Reply:
> Cannot we simplify to:
> 
> ```
> $ meziantou.validate-nuget-package ${{ env.NUGET_DIR }}/*.nupkg --excluded-rules Symbols
> ```
> 
> ??
> 
> Does `meziantou.validate-nuget-package` not accept multiple files at once?

Resolved in https://github.com/ButterCMS/buttercms-csharp/pull/40/commits/302382eee44defeb37f981f10c1f22263c6464d1